src/rs.c

```diff
@@ -377,9 +377,15 @@ hsk_rs_onrecv(
     goto fail;
   }
 
+  // Unbound's name resolution API expects a single null-terminated string.
+  // Since 0x00 is a valid byte mid-label in wire format we need to
+  // convert `req->name` to presentation format (i.e. "\000" for 0x00)
+  if (!hsk_dns_name_to_string(req->name, req->namestr))
```

Comment:
> Instead of adding this here, I believe it's more suitable to have it allocated only in hsk_rs_worker_resolve, where it's actually needed. There's no need to carry full string version just for unbound call.
> from req.h comment

Here we can also get rid of this part and only translate to string inside hsk_rs_worker_resolve.

Reply:
done in 1fffc2b831cdd53bc14c46d18122556f54848912